src/components/KMultiselect/KMultiselect.vue

```diff
@@ -0,0 +1,911 @@
+          {{ item ? item.label : '' }}
+        </KBadge>
+        <!-- Always render this badge even if it's hidden to ensure there will be
+             enough space to show it
+        -->
+        <KBadge
+          shape="rectangular"
+          hidden
+          class="mt-2 hidden-selection-count"
+        >
+          +{{ invisibleSelectedItemsStaging.length }}
+        </KBadge>
+      </div>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, ref, Ref, computed, watch, PropType, nextTick } from 'vue'
+import { v1 as uuidv1 } from 'uuid'
+import useUtilities from '@/composables/useUtilities'
+import KBadge from '@/components/KBadge/KBadge.vue'
+import KButton from '@/components/KButton/KButton.vue'
+import KIcon from '@/components/KIcon/KIcon.vue'
+import KInput from '@/components/KInput/KInput.vue'
+import KLabel from '@/components/KLabel/KLabel.vue'
+import KPop from '@/components/KPop/KPop.vue'
+import KToggle from '@/components/KToggle'
+import KMultiselectItem from '@/components/KMultiselect/KMultiselectItem.vue'
+
+const { getSizeFromString, cloneDeep } = useUtilities()
+const SELECTED_ITEMS_SINGLE_LINE_HEIGHT = 34
+
+const defaultKPopAttributes = {
+  hideCaret: true,
+  placement: 'bottomStart',
+  popoverTimeout: 0,
+  popoverClasses: 'k-multiselect-popover mt-0',
+}
+
+export interface MultiselectItem {
+  label: string
+  value: string
+  key?: string
+  selected?: boolean
+}
+
+export interface MultiselectFilterFnParams {
+  items: MultiselectItem[]
+  query: string
+}
+
+export default defineComponent({
+  name: 'KMultiselect',
+  components: {
+    KBadge,
+    KButton,
+    KIcon,
+    KInput,
+    KLabel,
+    KPop,
+    KMultiselectItem,
+    KToggle,
+  },
+  inheritAttrs: false,
+  props: {
+    modelValue: {
+      type: Array as PropType<string[]>,
+      default: () => [],
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    labelAttributes: {
+      type: Object,
+      default: () => ({}),
+    },
+    placeholder: {
+      type: String,
+      default: '',
+    },
+    kpopAttributes: {
+      type: Object,
+      default: () => ({
+        popoverClasses: '',
+      }),
+    },
+    dropdownMaxHeight: {
+      type: String,
+      default: '300',
+    },
+    /**
+     * The width of the multiselect and popover's min-width
+     */
+    width: {
+      type: String,
+      default: '',
+    },
+    /**
+     * Number of rows of selections to show when focused
+     */
+    selectionRowCount: {
+      type: Number,
+      default: 2,
+    },
+    /**
+     * Items are JSON objects with required 'label' and 'value'
+     * {
+     *  label: 'Item 1',
+     *  value: 'item1'
+     * }
+     */
+    items: {
+      type: Array as PropType<MultiselectItem[]>,
+      default: () => [],
+      // Items must have a label & value
+      validator: (items: MultiselectItem[]) => !items.length || items.some(i => i.label !== undefined && i.value !== undefined),
```

Comment:
This validator logic is a bit flawed (and it's also incorrect inside `KSelect`)

The `items.some` logic is saying that the prop value is valid if at least one of the array items has both a label and a value.

Instead, it needs to be `items.every` like here:
```suggestion
      validator: (items: MultiselectItem[]) => !items.length || items.every(i => i.label !== undefined && i.value !== undefined),
```

Can you make the exact same update to the `KSelect` `items` validator as well?

Reply:
Fixed in both